config/config.go

```diff
@@ -37,15 +41,30 @@ func NewConfig() *Config {
 	}
 }
 
-// Save stores the config to file
-func (c *Config) Save() error {
-	return save(ConfigPath, c)
+// Save stores the config to file.
+//
+// if the `path` is an empty string, it uses built-in user level config directory.
+func (c *Config) Save(path string) error {
+	// set default path
+	if path == "" {
+		path = dir.Path.ConfigForWrite(dir.UserLevel)
+	}
```

Comment:
Why do we need this if can we take this path from caller? 
Applicable for all load and save functions in config.go and keys.go.

Reply:
Because we have directory spec and it should be the default case (an empty path), and it is convenient for user to not specify the directory.

Reply:
Sure. Deleted.